charts/kyverno-policies/templates/validation/validate-pod-sec-ctxt-spire-server.yaml

```diff
@@ -0,0 +1,41 @@
+{{- $name := "validate-pod-sec-ctxt-spire-server" }}
+{{- include "kyverno-policy.supportedKyvernoCheck" (dict "top" . "ver" ">= 1.6.0-0") }}
+apiVersion: kyverno.io/v1
+kind: Policy
+metadata:
+  name: {{ $name }}
+  namespace: spire 
+  annotations:
+    policies.kyverno.io/title: Require runAsNonRoot, disallow Privilege Escalation and Privileged Containers
+    kyverno.io/kyverno-version: 1.6.0
+    policies.kyverno.io/subject: Pod
+    policies.kyverno.io/description: >-
+      Containers must be required to run as non-root users. Privilege escalation, such as via set-user-ID
+      or set-group-ID file mode, should not be allowed. Privileged mode disables most security mechanisms
+      and must not be allowed. This policy ensures `runAsNonRoot` is set to `true`, `allowPrivilegeEscalation`
+      field is set to `false` and pods do not call for privileged mode.
+spec:
+  background: true
+  validationFailureAction: audit
+  rules:
+  - name: container-security-context
+    match:
+      resources:
+        kinds:
+        - Pod
+        selector:
+          matchLabels:
+            app.kubernetes.io/name: spire-server
+    validate:
+      message: >-
+        Running as root is not allowed, Privilege escalation is disallowed and Privileged mode is disallowed. The fields
+        spec.containers[*].securityContext.runAsNonRoot must be set to `true`, spec.containers[*].securityContext.allowPrivilegeEscalation
+        must be set to `false` and spec.containers[*].securityContext.privileged must be unset or set to `false`.
+      pattern:
+        spec:
+          containers:
+          - (name): "*"
+            securityContext:
+              runAsNonRoot: true
```

Comment:
Notably for the runasuser stuff, we're mutating a set of policies, but not verifying the same policy. Thinking we may not need the runasuser/group mutations?

Reply:
No, we can add that too! Just to validate things, I thought runsAsNonRoot is enough. Should I add runasuser/group also?

Reply:
See recent comments. I think we should probably drop the runAsUser mutations.